### ISSUE TITLE
Build failed

### ISSUE THREAD

**Issue by dhwrwm** [zoontek/react-native-permissions#59]
error: method requestPermission in class PermissionsModule cannot be applied to given types;
    mPermissionsModule.requestPermission(permission, new PromiseImpl(resolve, reject));

**Comment by joncursi** [comment]
Same for me, after upgrading to RN 0.40.0

iOS builds fine, but Android build fails with this.

```
/Users/jcursi/Sites/joncursi/redbirdNative/node_modules/react-native-permissions/android/src/main/java/com/joshblour/reactnativepermissions/ReactNativePermissionsModule.java:96: error: method requestPermission in class PermissionsModule cannot be applied to given types;
    mPermissionsModule.requestPermission(permission, resolve, reject);
                      ^
  required: String,Promise
  found: String,Callback,Callback
  reason: actual and formal argument lists differ in length
1 error
:react-native-permissions:compileReleaseJavaWithJavac FAILED
```

**Comment by KarlosQ** [comment]
In iOS build failed too. Waiting to aproveed the pull request of danieldkim

**Comment by joncursi** [comment]
I believe these two issues are different. I am currently using a forked version of this library that includes new headers for (RN) 0.40.0 as provided in that PR and the build still fails on android with the above error.

**Comment by TalkingQuickly** [comment]
Also failing for me, @joncursi is the fork you're using https://github.com/joncursi/react-native-permissions by any chance?

**Comment by TalkingQuickly** [comment]
@joncursi I'm using your fork (https://github.com/joncursi/react-native-permissions) at the moment for the RN 0.40 headers fix, just starting to look into this Android build failure, have you come across any clues?

**Comment by joncursi** [comment]
Not yet. My fork only solves the (RN) 0.40.0 compatibility issues.

**Comment by TalkingQuickly** [comment]
OK for anyone who needs a combined version of the RN 0.40.0 fix from @joncursi and a fix for 

"method requestPermission in class PermissionsModule cannot be applied to given types;
mPermissionsModule.requestPermission(permission, new PromiseImpl(resolve, reject))"

on Android, i've put up https://github.com/joincatapult/react-native-permissions which is now working for me. 

Relevant commit is; https://github.com/joincatapult/react-native-permissions/commit/e1e25675f1618ccebe16e939781aa77848fa7979

**Comment by joncursi** [comment]
Thank you @TalkingQuickly ! Your commit is confirmed working on my end. I switched over from my fork to yours.

**Comment by TalkingQuickly** [comment]
Awesome @joncursi

**Comment by yonahforst** [comment]
merged pr #61 which should address this
